src/java/org/apache/cassandra/db/marshal/RedactionUtil.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.marshal;
+
+import java.nio.ByteBuffer;
+import javax.annotation.Nullable;
+
+/**
+ * Utility class for redacting sensitive data values while preserving some orientative size information.
+ * <p>
+ * This class provides methods to replace actual data values with a redacted placeholder ("?"), occasionally including
+ * size hints to help with debugging and troubleshooting without exposing the actual data content. No size hints will be
+ * included for values smaller than 100 bytes, or for values of fixed-size data types (e.g., int, UUID, timestamp).
+ * <p>
+ * Size hints are provided in logarithmic buckets (e.g., ">100B", ">1KiB", ">10KiB", ">100KiB") to give a rough
+ * indication of data size while maintaining privacy:
+ * <ul>
+ *   <li>Up to 100B: no size hint (just "?")</li>
+ *   <li>(100 B, 1 KiB]: "?[>100B]"</li>
+ *   <li>(1 KiB, 10 KiB]: "?[>1KiB]"</li>
+ *   <li>(10 KiB, 100 KiB]: "?[>10KiB]"</li>
+ *   <li>(100 KiB, 1 MiB]: "?[>100KiB]"</li>
+ *   <li>(1 MiB, 10 MiB]: "?[>1MiB]"</li>
+ *   <li>(10 MiB, 100 MiB]: "?[>10MiB]"</li>
+ *   <li>(100 MiB, 1 GiB]: "?[>100MiB]"</li>
+ *   <li>Over 1 GiB: "?[>1GiB]"</li>
+ * </ul>
+ */
+public class RedactionUtil
```

Comment:
I think we should make the utility class final and add an empty private constructor for this class.